### PR TITLE
EOS-23009: Enable current sensor monitoring for node server.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [ ] Unittest updated and all unittests are passing?

* [ ] Sanity/Self tests are updated (if applicable)?

* [ ] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [ ] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

**Review by sumedhak27 — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]
Current sensor data tested for current value reflections atleast ? Check if ipmitool allows to simulate event for current threshold value breach or not and if yes, test it out

File: low-level/sensors/impl/generic/node_hw.py [hunk 1]
```diff
@@ -1277,60 +1273,57 @@ def _parse_voltage_info(self, index, date, _time, sensor, sensor_num, event, sta
...
-#            }
-#        elif (alert_type in ['fault_resolved', 'miscellaneous'] or \
-#            (threshold.lower() in ['low', 'high'] and status.lower() == "deasserted")) \
-#                and sensor_name in self.faulty_resources:
-#            del self.faulty_resources[sensor_name]
-#
-#        self._send_json_msg(resource_type, alert_type, severity, info, specific_info)
-#        store.put(self.faulty_resources, self.faulty_resources_path)
+    def _parse_current_info(self, index, date, _time, sensor, sensor_num, event, status, is_last):
+
```
Comment (jjxsg) [inline low-level/sensors/impl/generic/node_hw.py:1277]: Please add docstring for function `_parse_current_info`

File: low-level/sensors/impl/generic/node_hw.py [hunk 2]
```diff
@@ -1277,60 +1273,57 @@ def _parse_voltage_info(self, index, date, _time, sensor, sensor_num, event, sta
...
+            severity_reader = SeverityReader()
+            severity = severity_reader.map_severity(alert_type)
+            if (
+                threshold.lower() in ['low', 'high'] and
+                    status.lower() == "deasserted"):
+                severity = "informational"
+        else:
+            alert_type = "miscellaneous"
+            severity = "informational"
+
```
Comment (jjxsg) [inline low-level/sensors/impl/generic/node_hw.py:1294]: Please remove extra empty line.

File: low-level/sensors/impl/generic/node_hw.py (hunk 1 above)
Comment (shriya-deshmukh) [reply low-level/sensors/impl/generic/node_hw.py]: done.

File: low-level/sensors/impl/generic/node_hw.py (hunk 2 above)
Comment (shriya-deshmukh) [reply low-level/sensors/impl/generic/node_hw.py]: Done.

**Comment by shriya-deshmukh** [comment]
> Current sensor data tested for current value reflections atleast ? Check if ipmitool allows to simulate event for current threshold value breach or not and if yes, test it out

Using 'ipmitool event' cmd we can simulate following event for current sensor, I have done the testing for same.
```
[root@iu19-r21 ~]# ipmitool event 'PS2 Curr Out %'
Finding sensor PS2 Curr Out %... ok
Sensor States:
  lnr : Lower Non-Recoverable
  lcr : Lower Critical
  lnc : Lower Non-Critical
  unc : Upper Non-Critical
  ucr : Upper Critical
  unr : Upper Non-Recoverable
```
eg: 
```
[root@iu19-r21 ~]# ipmitool event 'PS2 Curr Out %' lnr
Finding sensor PS2 Curr Out %... ok
   0 |  Pre-Init  |0000000000| Current #0x59 | Lower Non-recoverable going low  | Asserted
```

**Review by jjxsg — APPROVED** [review]

**Comment by cdeshmukh** [comment]
VM deployment was also good before branch was updated with latest main code, so i see this good to merge too

**Comment by cdeshmukh** [comment]
test this please

**Pull request merged**